zkvm/src/types.rs

```diff
@@ -301,11 +341,73 @@ impl Value {
 
 impl Expression {
     pub fn constant<S: Into<Scalar>>(a: S) -> Self {
-        let a: Scalar = a.into();
+        Expression::Constant(ScalarWitness::Scalar(a.into()))
```

Comment:
u64 is `Into<Scalar>`, and its integerness will be destroyed when you do `Expression::constant(1u64)`. Maybe use `Into<ScalarWitness>` instead of `Into<Scalar>` here?

Reply:
Yep, thank you